tests/lib/rules/no-unused-prop-types.js

```diff
@@ -784,54 +764,6 @@ ruleTester.run('no-unused-prop-types', rule, {
       ].join('\n')
     }, {
       code: [
-        'const statelessComponent = (props) => {',
```

Comment:
First of all for some reason the same test were copy-pasted  4 times. I had to remove this test all together because it was never validated correctly. Current implementation is not able to detect correctly nested definition of components (subRender technically is a react component). I don't think this is a common/idiomatic react style so I assume we shouldn't support it.